### PR TITLE
add wait before blocks placed

### PR DESCRIPTION
**Pull request by u9g**

### CODE REVIEW

**Comment by rom1504** [comment]
looks like you forgot some files ?

**Comment by u9g** [comment]
> looks like you forgot some files ?

yeah lol, its taking me a sec to do through web github :P

File: lib/cursor.js [hunk 1]
```diff
@@ -56,6 +57,7 @@ class Cursor {
     if (cursorBlock && this.buttons[2] && (!this.lastButtons[2] || cursorChanged)) {
       const vecArray = [new Vec3(0, -1, 0), new Vec3(0, 1, 0), new Vec3(0, 0, -1), new Vec3(0, 0, 1), new Vec3(-1, 0, 0), new Vec3(1, 0, 0)]
       const delta = cursorBlock.intersect.minus(cursorBlock.position)
+      await wait(200)
```
Comment (rom1504) [inline lib/cursor.js:60]: no that's not right, there should be a 200ms delay between places, not before the first place

File: lib/util.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+const sleep = ms => new Promise((resolve, reject) => setTimeout(resolve, ms))
+
+module.exports = { wait }
```
Comment (Moondarker) [inline lib/util.js:3]: wait != sleep :p

Comment (u9g) [reply lib/util.js:3]: omg i cant believe i did that

File: lib/cursor.js (hunk 1 above)
Comment (u9g) [reply lib/cursor.js:60]: ohh I see 1s

File: lib/util.js (hunk 2 above)
Comment (Moondarker) [reply lib/util.js]: Happens to best of us, but that's the exact reason I don't use web editor :)

File: lib/cursor.js (hunk 1 above)
Comment (Karang) [reply lib/cursor.js:60]: dont use async/await, it will mess the logic again

Comment (u9g) [reply lib/cursor.js:60]: how would await mess up logic?

Comment (rom1504) [reply lib/cursor.js:60]: see https://github.com/PrismarineJS/prismarine-web-client/commit/0506f5a70d6bbae23fde4eddc248b44385f7b8f7

Comment (Karang) [reply lib/cursor.js:60]: As you wrote it, i can already tell you it is already buggy: try spamming right-click very fast: if will call the update multiple time, enter in the if, wait 200ms in parallel and place N blocks at the same time.

**Comment by u9g** [comment]
still a wip, havent gotten around to it

**Comment by AwesomestCode** [comment]
Probably worth converting this to a draft for now.

File: lib/cursor.js (hunk 1 above)
Comment (u9g) [reply lib/cursor.js]: Changed how it works

**Comment by u9g** [comment]
closes https://github.com/PrismarineJS/prismarine-web-client/issues/159

File: lib/cursor.js [hunk 3]
```diff
@@ -41,8 +41,11 @@ class Cursor {
       if (document.pointerLockElement !== renderer.domElement) return
       this.buttons[e.button] = true
     })
+    bot.lastPlaced = 4 // ticks since last placed
```
Comment (Karang) [inline lib/cursor.js:44]: if this isn't needed in bot, you could have this variable in this. instead

**Review by Karang — COMMENTED** [review]
Looks much better ;)

File: lib/cursor.js (hunk 3 above)
Comment (u9g) [reply lib/cursor.js:44]: how can i make a variable across init and update event?

Comment (u9g) [reply lib/cursor.js:44]: constructor*

Comment (Karang) [reply lib/cursor.js:44]: it's a class so you just use `this.lastPlaced`

**Comment by AwesomestCode** [comment]
@u9g Is this ready for review?

**Comment by AwesomestCode** [comment]
please lint

**Review by rom1504 — APPROVED** [review]

**Pull request merged**